org.adempiere.base/src/org/compiere/model/MAttributeInstance.java

```diff
@@ -234,6 +244,24 @@ public int getValueInt()
 			return super.getValueNumber().intValue();
 	} // getValueInt
 
+	
+	/**
+	 * Set Value as keyNamePair
+	 * @param value
+	 */
+	public void setValueKeyNamePair(KeyNamePair value)
+	{
+		if (value.getName() == null)

```

Comment:
don't think the if is necessary here.

Reply:
@hengsin there may be some cases when display label is resolved to null, in that case it is good idea to default to _ID  value.

Reply:
that usually means there's issue to be fixed somewhere else and I don't think showing ID is useful for user (In a way it can mask the fact the ID doesn't resolve properly).

Reply:
Yes, we can go with assumption that name never null. this if block may be never traversed part. but here it was just safe exception handling.